Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h

```diff
@@ -106,13 +106,13 @@ struct Poisson_visitor {
 
 struct Poisson_skip_vertices {
   double ratio;
-  Random& m_random;
-  Poisson_skip_vertices(const double ratio, Random& random)
-    : ratio(ratio), m_random(random) {}
+  Poisson_skip_vertices(const double ratio = 0)
+    : ratio(ratio) {}
 
   template <typename Iterator>
-  bool operator()(Iterator) const {
-    return m_random.get_double() < ratio;
+  bool operator()(Iterator it) const {
+    // make the result deterministic for each iterator
+    return Random((std::size_t)(&*it)).get_double() < ratio;
```

Comment:
@lrineau : this is not something we should do, right?
@sgiraudot  what are you trying to achieve? you need different seeds per iterator? This won't be deterministic accross runs AFAIU.

Reply:
First, that does not seem efficient, to create an object of type `Random`, at each call, just to get a few random bits.

Second, I am surprised, because I think the previous version (before the patch) seemed deterministic, and then the second one seems to be non-deterministic, because that depends on the memory layout. And the comment says the contrary.

Reply:
The previous version produced a different range everytime a `for()` loop was used to iterator from begin to end, which created problems in the new version of `compute_average_spacing()` where the range is iterated twice. This patch was made to make the range consistently the same if iterated several times.

Reply:
can't you init the random with the size of the range as seed once?

Reply:
It's not a matter of initialization.
Do this:
```c++
for(const auto& : range)
  // something

for(const auto& : range)
  // something
```
and you don't get the same range on the second loop, no matter the initialization. Maybe I could use a trick to initialize a shared `Random` object when calling `begin()`…